### DISCUSSION TITLE
Questions about CaImAn

### DISCUSSION THREAD

**Discussion by LeoJuh** [flatironinstitute/CaImAn#1223] [General]
Hello everyone, 

I am currently trying to familiarize myself with the theory and code of CaIman. However, there are two major questions that I’ve been unable to find the answer to, and I would greatly appreciate it if someone could help me answer these.

1. It is stated that non-negative matrix factorization (NMF) can be used to demix the calcium traces of spatially overlapping neurons. While I understand roughly how NMF works, I don't really see how factorizing the t-series into the “W“ and “H“ matrixes enable a separation of signals from overlapping neurons.  Could anyone possibly shed light on how NMF accomplishes this demixing?
    
2. In the computation of dF/F, the primary trace (F) is formed by adding the residual signal (YrA) to the denoised trace (C). Subsequently, dF/F is calculated using the formula:

   dF/F = (F - Fd) / (Df + Fd)
                
I find the inclusion of the residual signal in this process somewhat counterintuitive. Wouldn't the calculation of dF/F be more accurate if the primary trace (F) consisted solely of the denoised signal (C)? What is the rationale behind incorporating the residual signal (YrA) into the trace?

**Comment by EricThomson** [comment]
Regarding 2, for `estimates.detrend_df_f()` there is a `use_residuals` keyword argument you can set to `False` if you want DFOF for your denoised solutions in C rather than the raw traces. I'm not sure why it defaults to `True`. Note however see Issue #1188 this keyword argument currently has no effect. I have fixed it in a recent PR. 

Regarding 1, the algorithm uses alternating least squares, first fitting A (the spatial components) and then fitting C (the temporal components). If there isn't too much correlation in the activity, then the fit to the temporal components can settle into quite different solutions even if there is a lot of overlap between the corresponding spatial components. This will fail if there is high correlation between overlapping components

**Reply by LeoJuh** [reply]
Ok thanks, that makes it clear to me!
